paymentsheet/src/test/java/com/stripe/android/paymentsheet/CustomerSessionPaymentSheetActivityTest.kt

```diff
@@ -0,0 +1,461 @@
+        }
+
+    @Test
+    fun `When single CBC card, has remove permissions, and can remove last PM, should be able to remove and edit`() =
+        runTest(
+            cards = listOf(
+                createCard(last4 = "4242", addCbcNetworks = true),
+            ),
+            isPaymentMethodRemoveEnabled = true,
+            allowsRemovalOfLastSavedPaymentMethod = true,
+        ) {
+            composeTestRule.onEditButton().performClick()
+
+            val cbcCard = composeTestRule.onSavedPaymentMethod(last4 = "4242")
+
+            cbcCard.assertIsEnabled()
+            cbcCard.assertHasModifyBadge()
+
+            composeTestRule.onModifyBadgeFor(last4 = "4242").performClick()
+
+            composeTestRule.onEditScreenRemoveButton().assertIsEnabled()
+        }
+
+    @Test
+    fun `When single CBC card but no remove permissions, can edit but not remove CBC card`() =
+        runTest(
+            cards = listOf(
+                createCard(last4 = "4242", addCbcNetworks = true),
+            ),
+            isPaymentMethodRemoveEnabled = false,
+            allowsRemovalOfLastSavedPaymentMethod = true,
+        ) {
+            composeTestRule.onEditButton().performClick()
+
+            val cbcCard = composeTestRule.onSavedPaymentMethod(last4 = "4242")
+
+            cbcCard.assertIsEnabled()
+            cbcCard.assertHasModifyBadge()
+
+            composeTestRule.onModifyBadgeFor(last4 = "4242").performClick()
+
+            composeTestRule.onEditScreenRemoveButton().assertDoesNotExist()
+        }
+
+    @Test
+    fun `When single CBC card has remove permissions but cannot remove last, can edit but not remove CBC card`() =
+        runTest(
+            cards = listOf(
+                createCard(last4 = "4242", addCbcNetworks = true),
+            ),
+            isPaymentMethodRemoveEnabled = true,
+            allowsRemovalOfLastSavedPaymentMethod = false,
+        ) {
+            composeTestRule.onEditButton().performClick()
+
+            val cbcCard = composeTestRule.onSavedPaymentMethod(last4 = "4242")
+
+            cbcCard.assertIsEnabled()
+            cbcCard.assertHasModifyBadge()
+
+            composeTestRule.onModifyBadgeFor(last4 = "4242").performClick()
+
+            composeTestRule.onEditScreenRemoveButton().assertDoesNotExist()
+        }
+
+    private fun runTest(
+        cards: List<PaymentMethod>,
+        isPaymentMethodRemoveEnabled: Boolean,
+        allowsRemovalOfLastSavedPaymentMethod: Boolean,
+        test: (PaymentSheetActivity) -> Unit,
+    ) {
+        networkRule.enqueue(
+            RequestMatchers.host("api.stripe.com"),
+            RequestMatchers.method("GET"),
+            RequestMatchers.path("/v1/elements/sessions"),
+        ) { response ->
+            response.setBody(createElementsSessionResponse(cards, isPaymentMethodRemoveEnabled))
+        }
+
+        val countDownLatch = CountDownLatch(1)
+
+        ActivityScenario.launch<PaymentSheetActivity>(
```

Comment:
Is there another thing we should do? Like close the scenario? Or is that handled here. I like testing new tests like this with [ShampooRule](https://gist.github.com/JakeWharton/7fe7deb1f7f4a795c120). And if you don't close things properly, it will just fail the tests due to that.